fairscale/optim/oss.py

```diff
@@ -109,25 +109,10 @@ def __init__(
         # Current default device is set by the parameters allocated to this rank
         self._device = list(self.per_device_params.keys())[0]
         self.buckets: Dict[torch.device, List[torch.Tensor]] = {}
+        self.buffer_max_size = broadcast_buffer_size
 
-        # Get the correct size for the buckets, cannot be bigger than the model
-        model_size = sum([p.numel() for p in self.param_to_rank.keys()])
```

Comment:
this was done once at construction time, but the logic was broken in that the buckets could be resized when deciding for the bucketing strategy